src/main/java/org/truffleruby/core/symbol/SymbolNodes.java

```diff
@@ -112,16 +115,18 @@ protected DynamicObject createProc(InternalMethod method, DynamicObject symbol)
                     "proc",
                     ArgumentDescriptor.ANON_REST,
                     false);
-
-            final RubyRootNode rootNode = new RubyRootNode(getContext(), sourceSection, new FrameDescriptor(nil()), sharedMethodInfo, Translator.sequence(sourceIndexLength, Arrays.asList(Translator.createCheckArityNode(Arity.AT_LEAST_ONE), new SymbolProcNode(Layouts.SYMBOL.getString(symbol)))));
+            Object[] args = RubyArguments.pack(null, null, method, null, nil(), null, new Object[0]);
+            final MaterializedFrame declarationFrame = Truffle.getRuntime().createMaterializedFrame(args, descriptor);
+            final RubyRootNode rootNode = new RubyRootNode(getContext(), sourceSection, new FrameDescriptor(nil()), sharedMethodInfo,
+                    Translator.sequence(sourceIndexLength, Arrays.asList(Translator.createCheckArityNode(Arity.AT_LEAST_ONE), new SymbolProcNode(Layouts.SYMBOL.getString(symbol)))));
 
             final CallTarget callTarget = Truffle.getRuntime().createCallTarget(rootNode);
 
             return ProcOperations.createRubyProc(
                     coreLibrary().getProcFactory(),
                     ProcType.PROC,
                     sharedMethodInfo,
-                    callTarget, callTarget, null,
+                    callTarget, callTarget, declarationFrame,
```

Comment:
Note that MRI doesn't allow to get the binding of a Symbol#to_proc:
```ruby
:foo.to_proc.binding
ArgumentError: Can't create Binding from C level Proc
```
But if it's useful for us, it seems fine to allow it.

Reply:
Yeah I noticed that, and I'm not sure it's a limitation we should strive to replicate. :-)

Reply:
Might be worth adding a comment about it to clarify (MRI doesn't handle it, but we do it so all Proc have a decl frame).

Reply:
Comment added, let me know if there's anything else you think needs changing.